website/docs/API.md

```diff
@@ -515,22 +517,94 @@ Defined as:
 ```jsx
 function waitFor<T>(
   expectation: () => T,
-  { timeout: number = 4500, interval: number = 50 }
+  { timeout: number = 1000, interval: number = 50 }
 ): Promise<T> {}
 ```
 
-Waits for non-deterministic periods of time until your element appears or times out. `waitFor` periodically calls `expectation` every `interval` milliseconds to determine whether the element appeared or not.
+Should be used when testing asynchronous code or code with timers. `waitFor` allows to wait for some expectation (e.g. the presence of an element on the screen). The `expectation` param is a callback that should throw until what you're waiting for has realized.
+
+For instance, this is how we'd wait for an element to appear:
+
+```tsx
+await waitFor(() => expect(screen.getByText('Banana ready').toBeTruthy()))
```

Comment:
No need to wrap `getBy*` queries in `expect` since `getBy` is already throwing in case of match failure. 
```suggestion
await waitFor(() => screen.getByText('Banana ready'))
```

Reply:
While this also works I find it to be less explicit regarding the intent of this line, with explicit expect I find tests to be easier to understand. This is also the approach recommended by Kent C. Dodds (https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-get-variants-as-assertions). If the documentation uses plain get queries then it is very likely that users will use them as is so I think tests in the documentation should mirror what we consider to be the best practices. The question is wether we think or not that this is the best practice, it's not if the upmost importance so if you feel strongly about it we can recommend using plain get queries

Reply:
maybe worth mentionning that in a real life scenario, you should use findBy?

Reply:
If feel like it would rather belong in findBy doc, wdyt ?